src/telemetry/telemetry.c

```diff
@@ -0,0 +1,290 @@
+#define MAX_REQUEST_SIZE	4096
+
+#define REQ_DB_UUID					"db_uuid"
+#define REQ_EXPORTED_DB_UUID		"exported_db_uuid"
+#define REQ_INSTALL_TIME			"installed_time"
+#define REQ_INSTALL_METHOD			"install_method"
+#define REQ_OS						"os_name"
+#define REQ_OS_VERSION				"os_version"
+#define REQ_OS_RELEASE				"os_release"
+#define REQ_PS_VERSION				"postgresql_version"
+#define REQ_TS_VERSION				"timescaledb_version"
+#define REQ_BUILD_OS				"build_os_name"
+#define REQ_BUILD_OS_VERSION		"build_os_version"
+#define REQ_DATA_VOLUME				"data_volume"
+#define REQ_NUM_HYPERTABLES			"num_hypertables"
+#define REQ_RELATED_EXTENSIONS		"related_extensions"
+
+#define PG_PROMETHEUS	"pg_prometheus"
+#define POSTGIS			"postgis"
+
+static const char *related_extensions[] = {PG_PROMETHEUS, POSTGIS};
+static const char *version_delimiter[3] = {".", ".", ""};
+
+/*
+ * Parse the JSON response from the TS endpoint. There should be a field
+ * called "current_timescaledb_version". Check this against the local
+ * version, and notify the user if it is behind.
+ */
+static void
+process_response(const char *endpoint_response)
+{
+	int			i;
+	char	   *curr_sub_version;
+	long		curr_sub_version_long;
+	const long	local_version[3] = {
+		strtol(TIMESCALEDB_MAJOR_VERSION, NULL, 10),
+		strtol(TIMESCALEDB_MINOR_VERSION, NULL, 10),
+	strtol(TIMESCALEDB_PATCH_VERSION, NULL, 10)};
+	Datum		version = DirectFunctionCall2(json_object_field_text,
+											  CStringGetTextDatum(endpoint_response),
+											  PointerGetDatum(cstring_to_text(TS_VERSION_JSON_FIELD)));
+	char	   *version_string = text_to_cstring(DatumGetTextPP(version));
+
+	if (version_string == NULL)
+		elog(ERROR, "could not get TimescaleDB version from server response");
+
+	/*
+	 * Now parse the version string. We expect format to be XX.XX.XX, and if
+	 * not, we error out
+	 */
+	for (i = 0; i < 3; i++)
+	{
+		curr_sub_version = strtok(i == 0 ? version_string : NULL, version_delimiter[i]);
+
+		if (curr_sub_version == NULL)
+			elog(ERROR, "ill-formatted TimescaleDB version from server response");
+
+		curr_sub_version_long = strtol(curr_sub_version, NULL, 10);
+
+		if (local_version[i] < curr_sub_version_long)
+		{
+			ereport(LOG, (errmsg("you are not running the most up-to-date version of TimescaleDB."),
+						  errhint("The most up-to-date version is %s, your version is %s",
+								  version_string, TIMESCALEDB_VERSION_MOD)));
+			return;
+		}
+		if (local_version[i] > curr_sub_version_long)
+			break;
+	}
+
+	/*
+	 * Put the successful version check in a lower logging level to avoid
+	 * clogging logs.
+	 */
+	elog(NOTICE, "you are running the most up-to-date version of TimescaleDB.");
+}
+
+static char *
+get_num_hypertables()
+{
+	StringInfo	buf = makeStringInfo();
+
+	appendStringInfo(buf, "%d", number_of_hypertables());
+	return buf->data;
+}
+
+static char *
+get_database_size()
+{
+	StringInfo	buf = makeStringInfo();
+	int64		data_size = DatumGetInt64(DirectFunctionCall1(pg_database_size_oid,
```

Comment:
I still think it's weird to get the full database size. I'd prefer if we summed the total bytes of hypertables from our hypertable_size function. It also seems like it's more useful information for us, no? (we don't care much about the size of the db, we care more about the size of our hypertables...)

Reply:
I guess I agree, or getting both infos. But I'll let others chime in.

Reply:
It was simpler to get the size of the database. I agree we should have total size of all hypertables, but seeing as a lot of people have hypertables that interact with non-hypertables, this is also good info. I have a note in the design doc that to remind us to fetch hypertable size in future iterations. I think this is fine for MVP.